src/routes/relay/entities/schema/sponsored-call.schema.helper.ts

```diff
@@ -298,16 +318,14 @@ interface CreateProxyWithNonceTransactionData {
 const decodeCreateProxyWithNonce = (
   encodedData: string,
 ): CreateProxyWithNonceTransactionData => {
-  const CREATE_PROXY_WITH_NONCE_FRAGMENT =
-    'function createProxyWithNonce(address _singleton, bytes memory initializer, uint256 saltNonce)';
-
-  const createProxyWithNonceInterface = new ethers.Interface([
-    CREATE_PROXY_WITH_NONCE_FRAGMENT,
-  ]);
+  const proxyFactoryInterface = Proxy_factory__factory.createInterface();
+  const createProxyWithNonceFragment = proxyFactoryInterface.getFunction(
+    'createProxyWithNonce',
+  );
 
   const [singleton, initializer, saltNonce] =
-    createProxyWithNonceInterface.decodeFunctionData(
-      CREATE_PROXY_WITH_NONCE_FRAGMENT,
+    proxyFactoryInterface.decodeFunctionData(
+      createProxyWithNonceFragment,
```

Comment:
I don't think we need the fragment here.

```suggestion
      'createProxyWithNonce',
```

Reply:
As above regarding type safety.